src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/NetCorePackageReferenceProject.cs

```diff
@@ -42,6 +44,8 @@ public class NetCorePackageReferenceProject : BuildIntegratedNuGetProject
 
         private readonly IProjectSystemCache _projectSystemCache;
         private readonly UnconfiguredProject _unconfiguredProject;
+        private List<(NuGetFramework, Dictionary<string, ProjectInstalledPackages>)> _installedPackages = new List<(NuGetFramework, Dictionary<string, ProjectInstalledPackages>)>();
```

Comment:
You might want to provide a comparer into the dictionary. 

Package Ids are case insensitive.